### ISSUE TITLE
Setup script requires outdated version of NodeJs

### ISSUE THREAD

**Issue by maxchehab** [Nordstrom/hello-retail-workshop#56]
It seems that the setup script requires specifically Node 4.0. 

Does this repository depend on this specific node version? If so, a better way to manage this requirement would be to use the `engines` parameter in `package.json`. Since it seems that this setup script is a wrapper for the command `nmp install` this would disable different versions of node being used upon installation.

An example of using the `engines` parameter goes as follows:
```json
// package.json
{
   "engines":{
      "node":"4.0"
   }
}

// .npmrc (in the root directory)
engine-strict=true
```

`.npmrc` is a config file/script that is executed for each `npm` command. `engine-strict=true` will force the correct engines specified `package.json` to be present.

**Comment by erikerikson** [comment]
I'm not certain where you are referring to but we previously required *at least* node 4.x though may have been a bit more careful based on the testing we did at the time originally we wrote this.  We attempt to be compatible with all active versions of node until end of life.  We have observed people having issues with recent 10.x versions but expect users to currently otherwise succeed with 6.x+.  There is a scheduled effort for updating and improving this documentation.
